swiftnav/src/coords/ellipsoid.rs

```diff
@@ -0,0 +1,221 @@
+use super::{AzimuthElevation, LLHRadians, ECEF, NED};
+use crate::math::compile_time_sqrt;
+use nalgebra::Matrix3;
+
+/// Ellipsoid
+///
+/// An ellipsoid can be defined in terms of the semi-major axis and a second
+/// parameter, here we choose to use the inverse flattening term. The other
+/// parameters are derived from these two values.
+pub trait Ellipsoid {
+    /// Semi-major axis of the Earth in meters.
+    const A: f64;
+    /// Inverse flattening of the Earth.
+    const IF: f64;
+
+    /// The flattening of the Earth.
+    const F: f64 = 1.0 / Self::IF;
+    /// Semi-minor axis of the Earth in meters.
+    const B: f64 = Self::A * (1.0 - Self::F);
+    /// Eccentricity of the Earth,  where e^2 = 2f - f^2
+    const E: f64 = compile_time_sqrt(2.0 * Self::F - Self::F * Self::F);
+
+    /// Convert a [`LLHRadians`] to [`ECEF`]
+    #[must_use]
+    fn llh2ecef(llh: &LLHRadians) -> ECEF {
```

Comment:
Can we remove the `a2b` type names here? I think calling it `from_` is a bit more idiomatic. The code would read like `ECEF::from(llc)` or `ECEF::from_llc(val)`, whatever you feel is better.